single_kernel_mongo/state/app_peer_state.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""The peer relation databag."""
+
+import json
+from enum import Enum
+
+from ops.model import Application, Relation
+from typing_extensions import override
+
+from single_kernel_mongo.config.literals import SECRETS_APP
+from single_kernel_mongo.core.structured_config import MongoDBRoles
+from single_kernel_mongo.lib.charms.data_platform_libs.v0.data_interfaces import (  # type: ignore
+    DataPeerData,
+)
+from single_kernel_mongo.state.abstract_state import AbstractRelationState
+
+
+class AppPeerDataKeys(str, Enum):
+    """Enum to access the app peer data keys."""
+
+    managed_users_key = "managed-users-key"
+    db_initialised = "db_initialised"
+    role = "role"
+    replica_set_hosts = "replica_set_hosts"
+    keyfile = "keyfile"
+    external_connectivity = "external-connectivity"
```

Comment:
Yes they should, Mehdi made a similar comment elsewhere I just didn't take the time yet to do it (also because I do not like at all what I did there and I hope to be able to do better.

Reply:
Resolve this when you get the chance (or if its done in another PR just resolve this comment :) )

Reply:
I'll do it in #7 because there were multiple changes to the structure later in development and it would make the rebases quite hard :)

Reply:
There f18caaa